### ISSUE TITLE
[github process] Why doesn't UR repo use "squash and merge" instead of merge all the PR's commits.

### ISSUE THREAD

**Issue by yingcong-wu** [oneapi-src/unified-runtime#1668]
In intel/llvm repo, "squash and merge" is used over simply merging the PR branch into main branch. This will help keep a simple and clear main branch. I wonder why UR does not do the same. It bothers me that when I look at the github's commit history page, different commits from different PRs are interleaved and it is hard to follow the total change flow.

**Comment by pbalcer** [comment]
I find having per-PR commit history useful, since it allows the author to divide his work into individual pieces that can be separately reviewed, tested and, if required, bisected. However, these benefits only materialize if we can maintain the discipline to keep the PR commits tidy. We haven't been able to do so, and, given the volume of PRs, trying to enforce this now would probably be difficult.

So while I prefer the current approach, I agree that "squash and merge" might be a better option going forward.

**Comment by yingcong-wu** [comment]
Okay, I get your point but that is a high bar to achieve. Also, I don't like merging because it does not clearly incidate when a change is introduced in github's commit history page. If we can keep the main branch linear, then that would be a problem(without those merge commit).

**Comment by kbenzie** [comment]
Development now happens in https://github.com/intel/llvm
